### PR TITLE
Fix default manifest file mapping (path to name)

### LINKED ISSUE #70: assets file map incorrect
clone this repository
https://github.com/p2227/react-starter-kit/

run 
```bash
npm i
npm run build-stats
```
i will got the `manifest.json`
```
{
  "2f751285.png": "logo-small.png",
  "8844262b.png": "logo-small@2x.png",
  "about.js": "about.73e7ed35.chunk.js",
  "about.js.map": "about.73e7ed35.chunk.js.map",
  "admin.js": "admin.82b2d7f7.chunk.js",
  "admin.js.map": "admin.82b2d7f7.chunk.js.map",
  "client.js": "client.30fe54c9.js",
  "client.js.map": "client.30fe54c9.js.map",
  "contact.js": "contact.3e31ace8.chunk.js",
  "contact.js.map": "contact.3e31ace8.chunk.js.map",
  "home.js": "home.88e579db.chunk.js",
  "home.js.map": "home.88e579db.chunk.js.map",
  "login.js": "login.a43f9088.chunk.js",
  "login.js.map": "login.a43f9088.chunk.js.map",
  "not-found.js": "not-found.1b11123c.chunk.js",
  "not-found.js.map": "not-found.1b11123c.chunk.js.map",
  "privacy.js": "privacy.a840f532.chunk.js",
  "privacy.js.map": "privacy.a840f532.chunk.js.map",
  "register.js": "register.f35a0a4f.chunk.js",
  "register.js.map": "register.f35a0a4f.chunk.js.map",
  "vendor.js": "vendor.84747063.js",
  "vendor.js.map": "vendor.84747063.js.map"
}
```

but i expect 
```
   "logo-small.png": "2f751285.png",
  "logo-small@2x.png": "8844262b.png",
```

node v8.2.1 , mac 10.12.5

### PR DESCRIPTION
**Pull request by frequin**
closes #70

### CODE REVIEW

**Comment by mastilver** [comment]
Thank you @frequin , I think you are right, I must have inverted something here...

But it's not in the `reduce`
Could you rebase on `1.x` branch and add a test so it doesn't happens in the future :)

**Comment by mastilver** [comment]
@frequin This is where the issue lives: https://github.com/danethurber/webpack-manifest-plugin/blob/43b48f1eab84d718439383fa52a683c79e237fab/lib/plugin.js#L88-L89

**Comment by frequin** [comment]
@mastilver rebased and fixed. I'll add a test asap.

**Comment by mastilver** [comment]
Great! :)

I think minimal test would be to copy https://github.com/danethurber/webpack-manifest-plugin/blob/43b48f1eab84d718439383fa52a683c79e237fab/spec/plugin.spec.js#L345-L363

and add `[hash]`

**Comment by frequin** [comment]
@mastilver I've added a test. Could you take a look if it seems right to you ?

File: spec/plugin.spec.js [hunk 1]
```diff
@@ -387,6 +387,28 @@ describe('ManifestPlugin', function() {
...
+            { test: /\.(txt)/, loader: 'file-loader?name=[hash].[ext]' },
+          ]
+        }
+      }, {}, function(manifest, stats) {
+        var assetOutputFilename = Object.keys(stats.compilation.assets)[0];
+
+        expect(manifest).toBeDefined();
+        expect(manifest).toEqual({
+          'main.js': 'main.js',
+          'file.txt': assetOutputFilename
```
Comment (mastilver) [inline spec/plugin.spec.js]: `'file.txt': stats.hash + '.txt'`

Comment (frequin) [inline spec/plugin.spec.js]: Well `stats.hash` is related to the bundle hash while `[hash]` from the file-loader name is related to the file content. So it does not fit.
Do you see another way ?

Comment (frequin) [inline spec/plugin.spec.js]: Or instead of a hash, I can use `file-loader?name=outputfile.[ext]` ? What do you think ?

Comment (mastilver) [inline spec/plugin.spec.js]: As long as key and value are different, I'm happy :)

**Comment by frequin** [comment]
@mastilver There it is :)

**Comment by mastilver** [comment]
Great, Thank you @frequin

**Pull request merged**